File: Sem4/MPP/cs/lab3/ConsoleApp1/Repository/Repository.cs
using Domain;

namespace Repository;

public interface Repository<TID, TEntity> where TEntity : Entity<TID>
{
    List<TEntity> GetAll();
    TEntity GetOne(TID id);
    void Add(TEntity entity);
    void Update(TEntity entity);
    void Delete(TEntity entity);
    
}

File: Sem4/MPP/java/lab3java/model/src/main/java/ubb/scs/mpp/Repository/Repository.java
package ubb.scs.mpp.Repository;

import ubb.scs.mpp.Domain.Entity;

public interface Repository<ID, E extends Entity<ID>> {

    E findOne(ID id) throws RepoException;

    Iterable<E> findAll() throws RepoException;

    E save(E entity) throws RepoException;

    E delete(ID id) throws RepoException;

    E update(E entity) throws RepoException;
    
}
